python/tests/test_parsing.py

```diff
@@ -54,3 +55,31 @@ def test_invalid_project_config(example_dir):
 def test_empty_project_config(example_dir):
     with pytest.raises(ValueError):
         parse_project_config(example_dir / "invalid" / "empty")
+
+
+def test_valid_circular_dependencies(example_dir):
+    project_config = parse_project_config(example_dir / "valid")
+    modules = project_config.modules
+    all_cycles: set[tuple[str, ...]] = set()
+    for module in modules:
+        visited: set[str] = set()
+        path: list[str] = list()
+        find_cycle(module, visited, path, modules, all_cycles)
+    assert all_cycles == set()
+
+
+def test_cycles_circular_dependencies(example_dir):
+    project_config = parse_project_config(example_dir / "cycles")
+    modules = project_config.modules
+    all_cycles: set[tuple[str, ...]] = set()
+    for module in modules:
+        visited: set[str] = set()
+        path: list[str] = list()
+        find_cycle(module, visited, path, modules, all_cycles)
+    assert all_cycles == {
+        ("domain_one", "domain_three", "domain_one"),
+        ("domain_three", "domain_one", "domain_three"),
+        ("domain_three", "domain_one", "domain_two", "domain_three"),
+        ("domain_two", "domain_three", "domain_one", "domain_two"),
+        ("domain_one", "domain_two", "domain_three", "domain_one"),
+    }
```

Comment:
Thanks for the tests!

This is not the output I would expect from finding cycles in a graph though, and would probably be confusing to a user. It seems like finding 'unique' cycles in general might best be done with `networkx`. This dependency was added just recently in #182. Here is some example code that shows the approach we should adopt:

```
import networkx as nx

def canonical_form(cycle):
    """Rotate cycle to start with the smallest element."""
    n = len(cycle)
    min_index = cycle.index(min(cycle))
    return cycle[min_index:] + cycle[:min_index]

def find_unique_cycles(...):
    # Build a networkx.DiGraph
    # use the code in show.py:generate_module_graph_dot_file as a model for this
    
    # Find all simple cycles in the graph
    all_cycles = list(nx.simple_cycles(G))
    
    # Normalize cycles to their canonical form
    canonical_cycles = {tuple(canonical_form(cycle)) for cycle in all_cycles}
    
    # Convert tuples back to lists
    unique_cycles = [list(cycle) for cycle in canonical_cycles]
    
    return unique_cycles
```